htmresearch/support/union_temporal_pooler_monitor_mixin.py

```diff
@@ -28,30 +28,180 @@
 import numpy
 
 from nupic.research.monitor_mixin.metric import Metric
-from nupic.research.monitor_mixin.monitor_mixin_base import MonitorMixinBase
+from htmresearch.support.spatial_pooler_monitor_mixin import SpatialPoolerMonitorMixin
+from htmresearch.algorithms.union_temporal_pooler import UnionTemporalPooler
 from nupic.research.monitor_mixin.plot import Plot
 from nupic.research.monitor_mixin.trace import (
   IndicesTrace, StringsTrace, BoolsTrace, MetricsTrace)
 
+from nupic.bindings.math import GetNTAReal
 
 
-class UnionTemporalPoolerMonitorMixin(MonitorMixinBase):
+realDType = GetNTAReal()
+uintType = "uint32"
+
+
+
+class UnionTemporalPoolerMonitorMixin(SpatialPoolerMonitorMixin):
   """
   Mixin for UnionTemporalPooler that stores a detailed history, for inspection and
   debugging.
   """
 
   def __init__(self, *args, **kwargs):
+    if "distanceMetric" in kwargs:
+      self._mmDistanceMetric = kwargs["distanceMetric"]
+      del kwargs["distanceMetric"]
+    else:
+      self._mmDistanceMetric = "pctOverlapLarger"
+
     super(UnionTemporalPoolerMonitorMixin, self).__init__(*args, **kwargs)
 
     self._mmResetActive = True  # First iteration is always a reset
+    self.mmClearHistory()
```

Comment:
mmClearHistory() initializes the arrays in _mmData, so instead of initializing them separately in __init__(), calling mmClearHistory() from __init__() minimizes repetition.